source/extensions/filters/network/redis_proxy/redirection_mgr_impl.cc

```diff
@@ -0,0 +1,59 @@
+#include "extensions/filters/network/redis_proxy/redirection_mgr_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace RedisProxy {
+
+bool RedirectionManagerImpl::onRedirection(const std::string& cluster_name) {
+  auto it = info_map_.find(cluster_name);
+  if (it != info_map_.end()) {
+    ClusterInfoSharedPtr info = it->second;
+    if (info.get()) {
+      info->tracker_->update();
+      if (info->tracker_->eventsPerMinute() >= info->redirects_per_minute_threshold_) {
+        uint64_t now = std::chrono::duration_cast<std::chrono::milliseconds>(
+                           time_source_.monotonicTime().time_since_epoch())
+                           .count();
+        uint64_t last_callback_time_ms = info->last_callback_time_ms_.load();
+        if (!last_callback_time_ms ||
+            (now >= (last_callback_time_ms + info->min_time_between_triggering_.count()))) {
+          if (info->last_callback_time_ms_.compare_exchange_strong(last_callback_time_ms, now)) {
+            main_thread_dispatcher_.post([=]() {
+              // Ensure that cluster is still active before calling callback.
+              auto map = cm_.clusters();
+              auto it = map.find(cluster_name);
+              if (it != map.end()) {
+                info->cb_();
```

Comment:
I think it's important to reset the tracker after refresh is done.  Otherwise we'll keep refreshing for 1 minute.

Reply:
new code up that uses a redirect count instead of rate, count is reset when the callback is posted and redirects during the minimum time after triggering are ignored (not counted at all). This should prevent over-doing topology refresh.